mocks/mocks.go

```diff
@@ -248,6 +248,10 @@ func (sa *StorageAuthorityReadOnly) GetCertificateStatus(_ context.Context, req
 	}
 }
 
+func (sa *StorageAuthorityReadOnly) SetCertificateStatusReady(ctx context.Context, req *sapb.Serial, _ ...grpc.CallOption) (*emptypb.Empty, error) {
+	return nil, berrors.InternalServerError("not implemented")
```

Comment:
I imagine this (and the similar errors/panics in other mocks) will be changing, when a unittest that sets the feature flag to `true` is added?

Reply:
I wound up concluding that this really wants an integration test. I could also potentially write a unittest as well; curious to hear what you think. If I did, these baseline mock impls will likely stay the same and I would implement another mock type that provides specific implementations that I want.

Reply:
That's a good point, and points in the direction it would be kinda nice for us to go: all of our ["baseline" mocks](https://github.com/letsencrypt/boulder/blob/914e971f15589b33689f005ea1015b381e3d2c9b/mocks/mocks.go#L252) could return errors like this, instead of `nil, nil`, to force proper mocking of the individual methods we need closer to the test site.

Reply:
Returning berrors.InternalServer would result in (grpc-go) `codes.Internal` which doesn't seem appropriate here:
> Internal errors. This means that some invariants expected by the underlying system have been broken. This error code is reserved for serious errors.

For mocks I think it's reasonable to use `codes.Unimplemented`:
> The operation is not implemented or is not supported/enabled in this service.

You could use it here like so:
```suggestion
	return nil, status.Error(codes.Unimplemented, "unimplemented mock")
```
or you could implement a new method for it inside of berrors.